lib/evaluation.py

```diff
@@ -0,0 +1,106 @@
+from sklearn.metrics import *
+from abc import ABC, abstractmethod
+
+
+class Evaluator(ABC):
+	
+	def __init__(self, y_true, y_pred):
+		self.y_true = y_true
+		self.y_pred = y_pred
+
+	def data_is_valid(self) -> bool:
+		return len(self.y_true) == len(self.y_pred)
+
+	@abstractmethod
+	def evaluate(self) -> dict:
+		pass
+
+
+class TabularEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred):
+		super().__init__(y_true, y_pred)
+
+	def evaluate(self) -> dict:
+		pass
+
+
+class RegressionEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred):
+		super().__init__(y_true, y_pred)
+
+	def evaluate(self) -> dict:
+		
+		if not self.data_is_valid():
+			raise ValueError('Invalid Data')
+		
+		scores = {
+				'rootMeanSquaredError': mean_squared_error(self.y_true, self.y_pred, squared=False),
+				'meanAbsoluteError': mean_absolute_error(self.y_true, self.y_pred),
+				'meanAbsolutePercentageError': 'Infinity' if any(self.y_true == 0 | self.y_pred == 0) else
+				mean_absolute_percentage_error(self.y_true, self.y_pred),
+				'rSquared': r2_score(self.y_true, self.y_pred),
+				'rootMeanSquaredLogError': 'NaN' if any(self.y_true < 0 | self.y_pred < 0) else mean_squared_log_error(
+						self.y_true, self.y_pred),
+		}
+		
+		return scores
+
+
+class ClassificationEvaluator(Evaluator):
```

Comment:
Please also provide a demo python script about how to use the evaluators here. The lib/test folder contains some examples but you don't need to follow them exactly.